controllers/vspheremachine_controller.go

```diff
@@ -191,61 +184,35 @@ func (r *VSphereMachineReconciler) reconcileVSphereMachineIPAddress(cluster *cap
 		}
 	}
 
-	if err := r.Patch(context.TODO(), vsphereMachine.DeepCopyObject(), dataPatch); err != nil {
-		return &ctrl.Result{}, errors.Wrapf(err, "failed to patch VSphereMachine %s", vsphereMachine.Name)
+	if err := r.Patch(context.TODO(), vSphereMachine.DeepCopyObject(), dataPatch); err != nil {
+		return &ctrl.Result{}, errors.Wrapf(err, "failed to patch VSphereMachine %s", vSphereMachine.Name)
 	}
 
 	log.V(0).Info("successfully reconciled IP address for VSphereMachine")
 
 	return &ctrl.Result{}, nil
 }
 
-func getMatchLabels(cli client.Client, clusterMeta metav1.ObjectMeta, vsphereMachine *infrav1.VSphereMachine, log logr.Logger) map[string]string {
+func (r *VSphereMachineReconciler) getIPPoolMatchLabels(cli client.Client, vSphereMachine *infrav1.VSphereMachine, log logr.Logger) map[string]string {
 	labels := map[string]string{}
 
-	//match labels are an aggregate of VSphereMachine labels & VSphereMachineTemplate labels
-	vmLabels := util.GetObjLabels(vsphereMachine)
-	for k, v := range vmLabels {
-		labels[k] = v
+	//match labels for the IPPool are retrieved from the VSphereMachineTemplate
+	vmTemplateName, ok := vSphereMachine.GetAnnotations()[capi.TemplateClonedFromNameAnnotation]
```

Comment:
when is this annotation added by capi? if recently added, what about existing clusters?

Reply:
yes, I verified that this was added in capi v0.3.7 - https://github.com/kubernetes-sigs/cluster-api/blob/846ca08db5939e5bf88104b2f34427e7316ee7b9/api/v1alpha3/common_types.go#L43. All clusters using capv-static-ip should be post v0.3.7. I'll update the compatibility matrix in the docs.